scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,670 @@
+        smaller the probability of acceptance. It has to be any negative value.
+    maxfun : int, optional
+        Soft limit for the number of objective function calls. If the
+        algorithm is in the middle of a local search, this number will be
+        exceeded, the algorithm will stop just after the local search is
+        done.
+    maxsteps : int, optional
+        The maximum number of gensa iterations will perform.
+    """
+    def __init__(self, fun, x0, bounds, args=(), seed=None,
+            temp_start=5230, qv=2.62, qa=-5.0, maxfun=1e7, maxsteps=500,
+            pure_sa=False):
+        self.fun = fun
+        self.args = args
+        self.pure_sa = pure_sa
+        if x0 is not None and not len(x0) == len(bounds):
+            raise ValueError('Bounds size does not match x0')
+        lu = zip(*bounds)
+        self._lower = np.array(lu[0])
+        self._upper = np.array(lu[1])
+        # Checking that bounds are consistent
+        if not np.all(self._lower < self._upper):
+            raise ValueError('Bounds are note consistent min < max')
+        # Initialization of RandomState for reproducible runs if seed provided
+        self._random_state = check_random_state(seed)
+        if x0 is None:
+            x0 = self._lower + self._random_state.random_sample(
+                    len(bounds)) * (self._upper - self._lower)
+        self.seed = seed
+        # Number of maximum sof iteration for local search
+        self.itsoftmax = x0.size * 6
+        # Size of the markov chain. Twice the dimension problem is a
+        # recommended value.
+        self.markov_length = x0.size * 2
+        # In case the real value of the global minimum is known
+        # it can be used as stopping criterion
+        self.know_real = False
+        self.real_threshold = -np.inf
+        # Maximum duration time of execution as a stopping criterion
+        # Default is unlimited duration
+        self.maxtime = np.inf
+        # Maximum number of function call that can be used a stopping criterion
+        self.maxfuncall = maxfun
+        # Maximum number of step (main iteration)  that ca be used as
+        # stopping criterion
+        self.maxsteps = maxsteps
+        # Minimum value of annealing temperature reached to perform
+        # re-annealing
+        self.temp_restart = 0.1
+        # Visiting distribution parameter
+        self.qv = qv
+        # Acceptance parameter value
+        self.qa = qa
+        # Initial temperature value for annealing
+        self.temp_start = temp_start
+        # Not yet implemented contraint function that would be used in the
+        # future
+        self.has_constraint = False
+        self.judge_constraint = None
+        self.factr = 1000
+        self.pgtol = 1.e-6
+        self.reps = 1.e-6
+
+        self._x = np.array(x0)
+        self._xrange = self._upper - self._lower
+        self._xbackup = np.array(self._x)
+        self._xmini = np.array(self._x)
```

Comment:
I spent a while confused by the suffix "mini".  If it is short for minimum then "min" would be clearer for me.  Or just the whole thing "minimum".  "mini" makes me think of miniature